client/state/sites/test/selectors.js

```diff
@@ -1047,4 +1073,405 @@ describe( 'selectors', () => {
+							options: {
+								jetpack_version: smallerVersion
+							}
+						}
+					}
+				}
+			};
+
+			const canUpdateFiles = siteCanUpdateFiles( state, siteId );
+			expect( canUpdateFiles ).to.equal( false );
+		} );
+
+		it( 'it should return `false` if is a multi-network site', () => {
+			const siteId = 77203074;
+
+			const state = {
+				sites: {
+					items: {
+						77203074: {
+							ID: siteId,
+							jetpack: true,
+							is_multi_network: true,
+						}
+					}
+				}
+			};
+
+			const canUpdateFiles = siteCanUpdateFiles( state, siteId );
+			expect( canUpdateFiles ).to.equal( false );
+		} );
+
+		it( 'it should return `false` if is not a main network site (urls don\'t match)', () => {
+			const siteId = 77203074;
+
+			const state = {
+				sites: {
+					items: {
+						77203074: {
+							ID: siteId,
+							jetpack: true,
+							URL: 'https://jetpacksite.me',
+							is_multisite: true,
+							options: {
+								is_multi_network: false,
+								unmapped_url: 'https://example.wordpress.com',
+								main_network_site: 'https://anotherexample.wordpress.com'
+							}
+						}
+					}
+				}
+			};
+
+			const canUpdateFiles = siteCanUpdateFiles( state, siteId );
+			expect( canUpdateFiles ).to.equal( false );
+		} );
+
+		it( 'it should return `false` if `disallow_file_mods` is disabled', () => {
+			const siteId = 77203074;
+
+			const state = {
+				sites: {
+					items: {
+						77203074: {
+							ID: siteId,
+							jetpack: true,
+							URL: 'https://jetpacksite.me',
+							is_multisite: true,
+							options: {
+								is_multi_network: false,
+								unmapped_url: 'https://example.wordpress.com',
+								main_network_site: 'https://example.wordpress.com',
+								file_mod_disabled: [
+									'disallow_file_mods',
+								]
+							}
+						}
+					}
+				}
+			};
+
+			const canUpdateFiles = siteCanUpdateFiles( state, siteId );
+			expect( canUpdateFiles ).to.equal( false );
+		} );
+
+		it( 'it should return `false` if `has_no_file_system_write_access` is disabled', () => {
+			const siteId = 77203074;
+
+			const state = {
+				sites: {
+					items: {
+						77203074: {
+							ID: siteId,
+							jetpack: true,
+							URL: 'https://jetpacksite.me',
+							is_multisite: true,
+							options: {
+								is_multi_network: false,
+								unmapped_url: 'https://example.wordpress.com',
+								main_network_site: 'https://example.wordpress.com',
+								file_mod_disabled: [
+									'has_no_file_system_write_access',
+								]
+							}
+						}
+					}
+				}
+			};
+
+			const canUpdateFiles = siteCanUpdateFiles( state, siteId );
+			expect( canUpdateFiles ).to.equal( false );
+		} );
+
+		it( 'it should return `true` for the site right configurations', () => {
+			const siteId = 77203074;
+
+			const state = {
+				sites: {
+					items: {
+						77203074: {
+							ID: siteId,
+							jetpack: true,
+							URL: 'https://jetpacksite.me',
+							is_multisite: true,
+							options: {
```

Comment:
Hmm, since the options don't include the jetpack version, shouldn't this test be failing?